### PR TITLE
[Doc] API documentation updates.

### PR DESCRIPTION
**Pull request by dan-zheng**
Update API docs, mostly in the TensorFlow module.

### CODE REVIEW

File: stdlib/public/TensorFlow/ShapedArray.swift [hunk 1]
```diff
@@ -345,26 +345,38 @@ internal extension ShapedArray where Scalar : _TensorFlowDataTypeCompatible {
 }
 
 public extension ShapedArray {
+  /// The number of dimensions of the `ShapedArray`.
```
Comment (rxwei) [inline stdlib/public/TensorFlow/ShapedArray.swift]: I haven't seen Swift API docs use a definite article followed by the type name to denote `self`. `Array`'s [`count`](https://developer.apple.com/documentation/swift/array/2943906-count) says "in the array". So perhaps we should do the same.

File: stdlib/public/TensorFlow/ShapedArray.swift [hunk 2]
```diff
@@ -487,6 +499,13 @@ extension ShapedArray : RandomAccessCollection, MutableCollection {
 }
 
 public extension ShapedArray {
+  /// Calls a closure with a pointer to the `ShapedArray`’s contiguous storage.
```
Comment (rxwei) [inline stdlib/public/TensorFlow/ShapedArray.swift]:
```suggestion
  /// Calls a closure with a pointer to the array’s contiguous storage.
```

File: stdlib/public/TensorFlow/ShapedArray.swift [hunk 3]
```diff
@@ -495,6 +514,14 @@ public extension ShapedArray {
     }
   }
 
+  /// Calls the given closure with a pointer to the `ShapedArray`’s mutable
```
Comment (rxwei) [inline stdlib/public/TensorFlow/ShapedArray.swift]:
```suggestion
  /// Calls the given closure with a pointer to the array’s mutable
```

File: stdlib/public/TensorFlow/ShapedArray.swift [hunk 4]
```diff
@@ -696,28 +722,39 @@ public extension ShapedArraySlice {
     return baseIndices.count
   }
 
+  /// The number of dimensions of the `ShapedArraySlice`.
   var rank: Int {
     return base.rank - indexingDepth
   }
 
+  /// The dimensions of the `ShapedArraySlice`.
```
Comment (rxwei) [inline stdlib/public/TensorFlow/ShapedArray.swift]: This can just be "shape" because it's pretty "well defined".

**Review by rxwei — APPROVED** [review]

File: stdlib/public/TensorFlow/ShapedArray.swift (hunk 1 above)
Comment (rxwei) [reply stdlib/public/TensorFlow/ShapedArray.swift]: You can make a decision. If it's to be changed, other places should follow as well.

Comment (dan-zheng) [reply stdlib/public/TensorFlow/ShapedArray.swift]: I chose to use "array" for now. "Shaped array" might be good too, to distinguish from normal `Array` and usages of "array".

Done in https://github.com/apple/swift/pull/23093/commits/7b40a3e955845cb5b670e230dea499b91c83f895.

File: stdlib/public/TensorFlow/ShapedArray.swift (hunk 4 above)
Comment (dan-zheng) [reply stdlib/public/TensorFlow/ShapedArray.swift]: Done in https://github.com/apple/swift/pull/23093/commits/3540dd5c23e763c7c29d5673f0db050580659617.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

**Pull request merged**